dll/shellext/zipfldr/CZipExtract.cpp

```diff
@@ -565,6 +594,57 @@ class CZipExtract :
         Close();
         return true;
     }
+
+    bool PrepareWriteError(HWND hDlg, LPCSTR path, HRESULT hr)
```

Comment:
You may be able to merge this function with the other one, and have the two different behaviour via an extra flag/boolean (that controls the initialization of `strText` and the `MessageBoxA` flags).